ledger-core/src/core/database/DatabaseSessionPool.cpp

```diff
@@ -55,6 +59,12 @@ namespace ledger {
                     session.set_log_stream(_logger);
             }
 
+#ifdef PG_SUPPORT
+            _type = std::dynamic_pointer_cast<PostgreSQLBackend>(backend) != nullptr ?
+                    api::DatabaseBackendType::POSTGRESQL : api::DatabaseBackendType::SQLITE3;
```

Comment:
This is okay for this version but potentially ill-designed, as if we at some time support three different backends, that will just be wrong.

Reply:
See LLC-570.